music21/musicxml/m21ToXml.py

```diff
@@ -3369,8 +3438,28 @@ def parseOneElement(self, obj):
             if parsedObject is False:
                 environLocal.printDebug(['did not convert object', obj])
 
-        for sp in postList:  # directions that follow the element
-            root.append(sp)
+        if not skipPrePostObjectSpanners:
+            for sp in postList:  # directions that follow the element
+                root.append(sp)
+
+    def _hasPrePostObjectSpanners(self, obj) -> bool:
+        '''
+        returns True if and only if:
+        (1) there are spanners related to the object that should appear before the object
+        to the <measure> tag, or (2) there are spanners related to the object that should
+        appear after the object in the measure tag.
+        '''
+        spannerBundle = self.spannerBundle.getBySpannedElement(obj)
+        if not spannerBundle:
+            return False
+
+        # this list of spanner classes must match the paramsSet keys in prePostObjectSpanners()
+        for m21spannerClass in ('Ottava', 'DynamicWedge', 'Line'):
```

Comment:
Can we use the class literals here? Even better would be to extract a class-level constant. (Take the existing dict, bring it up to the class, and then just iterate over the keys).

Reply:
I am trying to mimic the existing routine, which uses string literals.  I think I should not be randomly improving other code during this PR, but I would be happy to take a request for a new PR here.

Reply:
Existing routine == prePostObjectSpanners, new routine that mimics == _hasPrePostObjectSpanners

Reply:
Fair point!